optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/result/SubSingleBenchmarkResult.java

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright 2015 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.benchmark.impl.result;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import com.thoughtworks.xstream.annotations.XStreamOmitField;
+import org.optaplanner.benchmark.impl.measurement.ScoreDifferencePercentage;
+import org.optaplanner.benchmark.impl.report.BenchmarkReport;
+import org.optaplanner.benchmark.impl.statistic.ProblemStatistic;
+import org.optaplanner.benchmark.impl.statistic.PureSingleStatistic;
+import org.optaplanner.benchmark.impl.statistic.SingleStatistic;
+import org.optaplanner.benchmark.impl.statistic.StatisticType;
+import org.optaplanner.core.api.score.FeasibilityScore;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.impl.score.ScoreUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Represents 1 benchmark run for 1 Single Benchmark configuration for 1 {@link Solver} configuration for 1 problem
+ * instance (data set).
+ */
+@XStreamAlias("subSingleBenchmarkResult")
+public class SubSingleBenchmarkResult implements SolverProblemBenchmarkResult {
+
+    protected static final transient Logger logger = LoggerFactory.getLogger(SubSingleBenchmarkResult.class);
+
+    @XStreamOmitField
+    private SingleBenchmarkResult singleBenchmarkResult;
+
+    private int subSingleBenchmarkIndex = -1;
+
+    // @XStreamImplicit() // TODO FIXME
+    private List<PureSingleStatistic> subPureSingleStatisticList = null;
```

Comment:
Concurrency would fail, as SubSingles of the same Single could be run at the same time. That's why we're technically building new instances during copying from Single->SubSingle.